### PR TITLE
Clarify CODEOWNERS syntax exceptions

### LINKED ISSUE #1407: Underdocumented CODEOWNERS syntax
### What article on docs.github.com is affected?

https://docs.github.com/en/free-pro-team@latest/github/creating-cloning-and-archiving-repositories/about-code-owners#codeowners-syntax

### What part(s) of the article would you like to see updated?

The `CODEOWNERS syntax` seems to not document valid syntax properly. As we recently found on one of our repos, this is invalid:
```
/directory/to/code       @orgname/myteam-owners  # a comment
```
Because the inline comment syntax is unsupported. However, `.gitignore` files do seem to support these comment types. Are there any other exceptions to the `.gitignore` syntax that are not documented?

### Additional information

N/A

### PR DESCRIPTION
**Pull request by ajsirota**
Make it clearer that a CODEOWNERS file follows most, but not all, gitignore syntax rules to fix #1407 

### Why:
Fix #1407

### What's being changed:

see diff

### Check off the following:
- [x] All of the tests are passing.
- [x] I have reviewed my changes in staging. (look for the **deploy-to-heroku** link in your pull request, then click **View deployment**)
- [x] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [x] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

**Comment by janiceilene** [comment]
@ajs256 Thanks so much for opening a PR! I'll get this triaged for review 🌟

**Review by hubwriter — COMMENTED** [review]
@ajs256 - Thanks for contributing this. 👍 
I'd like to hold off on merging this just now and see if we can get some more detailed information on the CODEOWNERS syntax as a resolution for https://github.com/github/docs/issues/1407. Let's see where we get with that.

Thanks again!

**Comment by ajs256** [comment]
bump

**Comment by janiceilene** [comment]
Thanks for the update @ajs256! I'll get this re-triaged.

File: content/github/creating-cloning-and-archiving-repositories/about-code-owners.md
Comment (hubwriter) [inline content/github/creating-cloning-and-archiving-repositories/about-code-owners.md:87]: Previously we didn't use the dot in `gitignore` so I suggest removing it here for consistency.
```suggestion
There are some syntax rules for gitignore files that do not work in CODEOWNERS files:
```

**Review by hubwriter — COMMENTED** [review]
@ajs256 - Thanks for making this update, based on the information from @janiceilene 

I like the way you've arranged the information. It's very clear.
Just one tiny suggestion: removal of a "."

If you agree and make this (or an alternative) change, I'll merge this. 
Thanks again for pursuing and implementing this improvement. 
Much appreciated. 👍

**Review by hubwriter — APPROVED** [review]

**Pull request merged**
